packages/editor/src/components/DataSource/ApiForm/ApiForm.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import { ComponentSchema } from '@sunmao-ui/core';
+import { FetchTraitPropertiesSchema } from '@sunmao-ui/runtime';
+import { Static } from '@sinclair/typebox';
+import {
+  VStack,
+  HStack,
+  Box,
+  IconButton,
+  Flex,
+  Text,
+  Tabs,
+  TabPanels,
+  TabPanel,
+  TabList,
+  Tab,
+  Select,
+  Input,
+  Button,
+  CloseButton,
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionPanel,
+  AccordionIcon,
+  Spinner,
+} from '@chakra-ui/react';
+import { EditIcon } from '@chakra-ui/icons';
+import { useFormik } from 'formik';
+import { Basic } from './Basic';
+import { Headers as HeadersForm } from './Headers';
+import { Params } from './Params';
+import { Body } from './Body';
+import { Response } from './Response';
+import { Error as ErrorInfo } from './Error';
+import { EditorServices } from '../../../types';
+import { genOperation } from '../../../operations';
+
+interface Props {
+  api: ComponentSchema;
+  services: EditorServices;
+  store: Record<string, any>;
+  className: string;
+}
+
+const METHODS = ['get', 'post', 'put', 'delete', 'patch'];
+
+export const ApiForm: React.FC<Props> = props => {
+  const { api, services, store, className } = props;
+  const { editorStore } = services;
+  const [isEditing, setIsEditing] = useState(false);
+  const [name, setName] = useState(api.id);
+  const { registry, eventBus } = services;
+  const traitIndex = api.traits.findIndex(({ type }) => type === 'core/v1/fetch');
+  const trait = api.traits[traitIndex];
+  const formik = useFormik({
+    initialValues: {
+      ...(trait?.properties as Static<typeof FetchTraitPropertiesSchema>),
+    },
+    onSubmit: values => {
+      eventBus.send(
+        'operation',
+        genOperation(registry, 'modifyTraitProperty', {
+          componentId: api.id,
+          traitIndex: traitIndex,
+          properties: values,
+        })
+      );
+    },
+  });
+  const { values } = formik;
+  const result = useMemo(() => {
+    return store[api.id]?.fetch ?? {};
+  }, [store[api.id], api.id]);
+  const data = useMemo(() => {
+    return result.data instanceof Object
+      ? JSON.stringify(result.data, null, 2)
+      : String(result.data);
+  }, [result.data]);
+
+  const onFetch = async () => {
+    services.apiService.send('uiMethod', {
+      componentId: api.id,
+      name: 'triggerFetch',
+      parameters: {},
+    });
+  };
+  const onNameInputBlur = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+
+    if (value) {
+      if (value !== api.id) {
+        editorStore.changeDataSourceName(api, value);
+      }
+      setIsEditing(false);
+    }
+  };
+
+  useEffect(() => {
+    formik.setValues({
+      ...(trait?.properties as Static<typeof FetchTraitPropertiesSchema>),
+    });
+  }, [trait.properties]);
+  useEffect(() => {
+    if (api.id) {
+      setName(api.id);
+    }
+  }, [api.id]);
+
+  return (
```

Comment:
This jsx is too long. Can we split it to several parts?

Reply:
Already split the response result and error into another component. Other elements belong to the `ApiForm` component, maybe they don't need to split anymore.